tests/test_cli.py

```diff
@@ -68,15 +78,24 @@ def assert_metadata_stripped(filepath, on_windows=RUNNING_ON_WINDOWS):
 
 
 @pytest.mark.skipif(RUNNING_ON_WINDOWS, reason='xattr does not work on Windows')
-def test_process_image_full(image_with_metadata, monkeypatch):
+def test_process_image_full(image_with_metadata, monkeypatch, recwarn):
     """Test that cli.process_image() removes EXIF and extended attributes."""
+
     assert_metadata_stripped(image_with_metadata)
 
+    # Unremovable attributes may not be present in all system setups.
+    # This is to assert the warning message if the user has such system configurations.
+    if recwarn:
+        message = str(recwarn[0].message)
+        assert message.startswith('Extended attributes')
+        assert message.endswith('cannot be removed.')
```

Comment:
It looks like the current test code does not ever get here, so it is marked as not covered.

Reply:
This will only be run on the machine that has this behavior.

This is the result of `pytest --cov` on my local.

```
---------- coverage: platform darwin, python 3.12.7-final-0 ----------
Name                            Stmts   Miss  Cover   Missing
-------------------------------------------------------------
src/exif_stripper/__init__.py      32      0   100%
src/exif_stripper/cli.py           14      0   100%
tests/test_cli.py                  76      1    99%   62
-------------------------------------------------------------
TOTAL                             122      1    99%
```

or we might need to rewrite the functions to make it more easy to mock this behavior

Reply:
The real problem is the inconsistent behavior of the extended attributes. I am just going to exclude those lines from coverage in the `tests/` directory for now, so we can merge this.

Reply:
Yep, that's indeed a troubling issue 🤔 not sure we can do much on that though. Regarding the test case, I'm thinking of refactoring the functions a bit. I'm also interested in https://github.com/stefmolin/exif-stripper/issues/37, but might need to wait till I have more bandwidth 👀